src/pass/llvmgen.cpp

```diff
@@ -357,18 +359,29 @@ Value* LLVMGen::visit(Locate& locate)
 
 Value* LLVMGen::visit(FetchDataPtr& fetch_data_ptr)
 {
-    auto vec_val = eval(fetch_data_ptr.vec);
-    auto idx_val = eval(fetch_data_ptr.idx);
-    auto col_val = ConstantInt::get(lltype(types::UINT32), fetch_data_ptr.col);
+    if (fetch_data_ptr.idx == nullptr) fetch_data_ptr.idx = _idx(0);
```

Comment:
why is idx ever null?

Reply:
If you want to keep it like this, I will clean this up.